js/api/firebaseHelper.js

```diff
@@ -169,6 +169,102 @@ export function writeNewUserGlobal(userID, global) {
       console.log("User info written!");
     });
 }
+export function writeNewUserName(userID, name) {
+  database
+    .collection("users")
+    .doc(userID)
+    .set(
+      {
+        name: name
+      },
+      {
+        merge: true
+      }
+    )
+    .then(function() {
+      console.log("User info written!");
+    });
+}
+export function writeNewUserProjects(userID, projects) {
```

Comment:
Rewrite your `writeNewUser...` functions into a single function using `Promise.all`:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Pass your the user ID and the object of form values into this function and use the values in your Firebase methods where needed.